design/cdk-bootstrap.md

```diff
@@ -0,0 +1,199 @@
+# cdk-bootstrap
+
+`cdk-bootstrap` is a tool in the AWS CDK toolchain responsible for populating a given environment
+(that is, a combination of AWS account and region)
+with resources required by the CDK to perform deployments into that environment.
+It's a successor to the `cdk bootstrap` sub-command that is part of the `@aws-cdk/core` package,
+and is needed because of the increased bootstrapping requirements present in the
+["Continuous delivery for CDK apps"](https://github.com/aws/aws-cdk/pull/3437) design document.
+
+## Command-line interface
+
+Invoking the `bootstrap` tool looks something like this:
+
+```shell
+$ cdk-bootstrap [--trust accountId[,otherAccountId]...] aws:://account/region
```

Comment:
It should be possible to run this non-interactively. This means that there should be a way to specify the deployment policy here. I would argue that in the first version, we should just require `--policy X` and not implement the interactive menu, which we can add later.

Reply:
We should also support passing in a JSON file for the policy, not just managed policies.

So, `--policy NAME|FILE` I guess

Reply:
> We should also support passing in a JSON file for the policy, not just managed policies.

Why though? Is this feature worth it? Can't it be expressed by creating a ManagedPolicy, and then passing its ARN?

Reply:
OK, you are right. Let's just start with the name of the managed policy and see if we need to get more fancy in the future. Since we allow allow people to pass in the full template, they could embed a custom policy doc themselves if they want.